PR: Support for Binance RPC

weboluchi opened the pull request.

leoneparise commented on packages/rpc/src/binance/BinanceRPC.spec.ts:
```diff
@@ -0,0 +1,71 @@
+    
+    let rpc: BinanceRPC;
+    let address: string;
+    let binance_RPC_URL: string
```
What is `binance_RPC_URL`? Why not just create BiannceRPC with the url? Please use camelcase naming convetion: `binanceRpcUrl` is better.

leoneparise commented on packages/rpc/src/binance/models/BinanceNodeInfo.ts:
```diff
@@ -0,0 +1,45 @@
+ 
+class NodeInfo {
+    network: String
+    listen_addr: String
```
Please use camelcase instead of snake case. To transform the data into camelcase we use `@Expose` annotation. In this case we should have:
```
@Expose({ name: 'listen_addr' })
listenAddr: string

leoneparise commented on packages/rpc/src/binance/models/BinanceNodeInfo.ts:
```diff
@@ -0,0 +1,45 @@
+import BigNumber from 'bignumber.js';
+ 
+class NodeInfo {
+    network: String
```
`string` should be lowercased. Please review all string type defintion in the code.

leoneparise commented:
@weboluchi I added some comments. The rest is good to me. Once you resolve them, I'll merge your PR

weboluchi commented:
> @weboluchi I added some comments. The rest is good to me. Once you resolve them, I'll merge your PR

I have implemented and resolved all comments. Thank you

dpereskokov commented on packages/rpc/src/binance/BinanceRPC.spec.ts:
```diff
@@ -0,0 +1,71 @@
+
+    beforeEach(function() {
+        require('dotenv').config({ path: __dirname + '/../.env' });
+        binanceRpcUrl = getEnv('BinanceRPC');
```
One more thing regarding naming convention. Please use uppercased env names. Like you may find in other packages. `BinanceRPC ` should be `BINANCE_RPC_URL`.

dpereskokov commented on packages/rpc/src/binance/BinanceRPC.test.ts:
```diff
@@ -0,0 +1,69 @@
+
+    beforeEach(function() {
+        require('dotenv').config({ path: __dirname + '/../.env' });
+        binanceRpcUrl = getEnv('BinanceRPC')
```
same here

dpereskokov commented on packages/rpc/src/binance/BinanceRPC.ts:
```diff
@@ -0,0 +1,55 @@
+        return plainToClass(BinanceMarket, data);
+    }
+
+    async getNodeInfo(limit?: number, offset?: number): Promise<BinanceNodeInfo> {
```
Params seem to be redundant

dpereskokov commented:
> I have effected these changes

Please check them once again. I see no changes in the PR.

dpereskokov commented:
LGTM 🚢

weboluchi commented:
> > I have effected these changes
> 
> Please check them once again. I see no changes in the PR.

Thank you

> LGTM 🚢

Ok, thank you

leoneparise commented:
Lets shiiip it!
